core/src/main/java/com/netflix/iceberg/TableMetadataParser.java

```diff
@@ -66,7 +69,10 @@ public static String toJson(TableMetadata metadata) {
   }
 
   public static void write(TableMetadata metadata, OutputFile outputFile) {
-    try (OutputStreamWriter writer = new OutputStreamWriter(outputFile.create())) {
+    try (OutputStreamWriter writer = new OutputStreamWriter(
+            shouldCompressMetadata() ?
```

Comment:
I think it would be cleaner if this checked `outputFile.location()` to see if it ends in `.gz` and should compress. Otherwise, the logic is split in two places: one that determines the filename and one that determines whether the stream is compressed.

And that way, `shouldCompressMetadata` could be in `BaseMetastoreTableOperations` and doesn't need to be public.

Reply:
Actually, `shouldCompressMetadata` is in two places. Maybe it should actually be in an `IcebergEnvironment` class instead, along with the settings in `ThreadPools`.

Reply:
I just added a single helper method called `getFileExtension` that all TableOperation implementation use. Let me know if you think it still makes sense to just add a helper method pull the property from config.

Reply:
I think that the code in `TableMetadataParser` looks good and is correct to use the file extension when deciding whether to compress while writing or whether to read with decompression.

I keep thinking about the `getFileExtension` helper, and I think it is going to be difficult to set a Java system property to configure whether we are using compression, at least for Spark. Since we have a `Configuration` in both `BaseMetastoreTableOperations` and in `HadoopTableOperations`, I think it makes sense to pull configuration from it. Especially since we already have well-known ways to set values in Hadoop configs.

Reply:
Moved the config to a different class and using Configuration instead of System properties now.